app/models/spree/gateway/braintree_gateway_decorator.rb

```diff
@@ -45,10 +47,14 @@ def authorize money, credit_card, options={}
       # Include device data if present
       options[:device_data] = credit_card.device_data
 
+      current_merchant_account_id = BraintreePresenter.new(credit_card.payment_method).merchant_account_id(money)
+      options[:merchant_account_id] = current_merchant_account_id
+      credit_card.payment_method.preferences[:merchant_account_id] = current_merchant_account_id
+
       if (money.to_f / 100) >= credit_card.payment_method.preferred_three_d_threshold
         # Is expensive enough that three_d_secure should be used
-        options[:payment_method_nonce] = true
-        options[:three_d_secure] = {required: true}
+        options[:payment_method_nonce] = credit_card.encrypted_data
+        options[:three_d_secure] = { required: true }
```

Comment:
Was doing some work on the payment stuff and saw that this was changed from true to the `encrypted_data`. Your second link is the right code path.

A few lines below this you will see where that encrypted_data is passed in the second argument to authorize. This is the same value as `credit_card_or_vault_id` in your link. So basically the only requirement is that we make that option be true.

As you note, having a string doesn't hurt as this also evaluates to true. With this updated code basically the option starts as the encrypted_data, then the gateway updates it to that same value.